src/ray/core_worker/task_interface.h

```diff
@@ -43,45 +43,109 @@ struct ActorCreationOptions {
 /// A handle to an actor.
 class ActorHandle {
  public:
-  ActorHandle(const ActorID &actor_id, const ActorHandleID &actor_handle_id)
-      : actor_id_(actor_id),
-        actor_handle_id_(actor_handle_id),
-        actor_cursor_(ObjectID::FromBinary(actor_id.Binary())),
-        task_counter_(0) {}
+  ActorHandle(const ActorID &actor_id, const ActorHandleID &actor_handle_id,
+              const ray::rpc::Language actor_language,
+              const std::vector<std::string> &actor_creation_task_function_descriptor) {
+    inner_.set_actor_id(actor_id.Data(), actor_id.Size());
+    inner_.set_actor_handle_id(actor_handle_id.Data(), actor_handle_id.Size());
+    inner_.set_actor_language((int)actor_language);
+    *inner_.mutable_actor_creation_task_function_descriptor() = {
+        actor_creation_task_function_descriptor.begin(),
+        actor_creation_task_function_descriptor.end()};
+    inner_.set_actor_cursor(actor_id.Data(), actor_id.Size());
+  }
 
   /// ID of the actor.
-  const ray::ActorID &ActorID() const { return actor_id_; };
+  const ray::ActorID ActorID() const { return ActorID::FromBinary(inner_.actor_id()); };
 
   /// ID of this actor handle.
-  const ray::ActorHandleID &ActorHandleID() const { return actor_handle_id_; };
+  const ray::ActorHandleID ActorHandleID() const {
+    return ActorHandleID::FromBinary(inner_.actor_handle_id());
+  };
+
+  /// Language of the actor.
+  const ray::rpc::Language ActorLanguage() const {
+    return (ray::rpc::Language)inner_.actor_language();
+  };
+  // Function descriptor of actor creation task.
+  const std::vector<std::string> ActorCreationTaskFunctionDescriptor() const {
+    return ray::rpc::VectorFromProtobuf(inner_.actor_creation_task_function_descriptor());
+  };
+
+  /// The unique id of the last return of the last task.
+  /// It's used as a dependency for the next task.
+  const ObjectID ActorCursor() const {
+    return ObjectID::FromBinary(inner_.actor_cursor());
+  };
+
+  /// The number of tasks that have been invoked on this actor.
+  const int64_t TaskCounter() const { return inner_.task_counter(); };
+
+  /// The number of times that this actor handle has been forked.
+  /// It's used to make sure ids of actor handles are unique.
+  const int64_t NumForks() const { return inner_.num_forks(); };
+
+  std::unique_ptr<ActorHandle> Fork() {
```

Comment:
Why does this return a `unique_ptr`, instead of just an object?

Reply:
Done.